lib/capistrano/tasks/sidekiq.rake

```diff
@@ -207,12 +207,8 @@ namespace :sidekiq do
   end
 
   def pid_files
-    sidekiq_roles = Array(fetch(:sidekiq_roles)).dup
-    sidekiq_roles.select! { |role| host.roles.include?(role) }
-    sidekiq_roles.flat_map do |role|
-      processes = fetch(:"#{ role }_processes") || fetch(:sidekiq_processes)
-      Array.new(processes) { |idx| fetch(:sidekiq_pid).gsub(/\.pid$/, "-#{idx}.pid") }
-    end
+    processes = fetch(:sidekiq_options_per_process) ? options_for_server(host).count : fetch(:sidekiq_processes)
```

Comment:
Seems this method is just creating an array of pid files with size of the array based on the number of processes the server will have.  Since we have the new `options_for_server` method, we can eliminate some of the original code here.